app/liquid/liquid_renderer.rb

```diff
@@ -44,6 +50,39 @@ def images
 
   private
 
+  def member_hash
+    return nil if @member.blank?
+    values = @member.attributes.symbolize_keys
+    values[:welcome_name] = [values[:first_name], values[:last_name]].join(' ')
+    values[:welcome_name] = values[:email] if values[:welcome_name].blank?
+    values
+  end
+
+  def outstanding_fields(plugin_data)
+    isolate_from_plugin_data(plugin_data, :outstanding_fields)
+  end
+
+  def donation_bands(plugin_data)
+    isolate_from_plugin_data(plugin_data, :donation_bands).first
+  end
+
+  def isolate_from_plugin_data(plugin_data, field)
+    plugin_values = plugin_data.deep_symbolize_keys[:plugins].values().map(&:values).flatten
+    plugin_values.map{|plugin| plugin[field]}.flatten.compact
+  end
+
+  def location
+    return @location if @location.blank?
+    if @member.present? && @member.country.present? && @member.country.length == 2
+      country_code = @member.country
+    else
+      country_code = @location.country_code
+    end
```

Comment:
there's often no recognized member (new person coming to the site). we currently don't validate the country at all, and it was quicker to write a conditional than a validation and tests for member. could be more concise though.